src/routes/subscriptions.rs

```diff
@@ -129,27 +71,34 @@ pub async fn subscribe(
     State(app_state): State<Arc<AppState>>,
     Form(form): Form<FormData>,
 ) -> Result<impl IntoResponse, SubscribeError> {
-    let new_subscriber = form.try_into()?;
-    let mut transaction = app_state.pool.begin().await.map_err(SubscribeError::Pool)?;
+    let new_subscriber = form.try_into().map_err(SubscribeError::Validation)?;
+    let mut transaction = app_state
+        .pool
+        .begin()
+        .await
+        .context("Failed to acquire a Postgres connection from the pool.")?;
```

Comment:
`anyhow` provides a function `context` that lets us chain the custom `Display` message so we dont have to implement that trait ourselves